src/index.js

```diff
@@ -1,45 +1,195 @@
+
+ ${displayItem(taskLs)}`;
+
+  clearFields();
+};
+
+// Create new task
+const addItem = (description) => {
+  const anothertask = new List(description, taskLs.length + 1, false);
+  taskLs.push(anothertask);
+  updateLocalStorage();
+  showList();
+};
+// to edit
+
+const form = document.querySelector('.form');
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const task = document.querySelector('.taskInput').value;
+  addItem(task);
+});
+
+const deleted = (pop) => {
+  const removeLss = JSON.parse(localStorage.getItem('mylist'));
+  removeLss.forEach((element, index) => {
+    if (element.index == pop) {
+      removeLss.splice(index, 1);
+      localStorage.setItem('mylist', JSON.stringify(removeLss));
+
+      console.log(' ');
+    }
+  });
+  localStorage.setItem('mylist', JSON.stringify(removeLss));
+  location.reload();
+};
+// delete function
+
+const deleteItem = (ele) => {
+  if (ele.classList.contains('delete')) {
+    ele.parentElement.parentElement.parentElement.remove();
+    // removeItem();}
+    const pop = ele.parentElement.parentElement.classList.value;
+
+    deleted(pop);
+  }
+};
+
+// delete item
+const toDelete = document.querySelector('.list-content');
+if (!toDelete) {
+  console.log(' ');
+} else {
+  document.querySelector('.list-content').addEventListener('click', (e) => {
+    if (!e) {
+      console.log(' ');
+    } else {
+      deleteItem(e.target);
+    }
+  });
+}
+
+// event to delete from ls
+
+// editing
+const edits = (description, index, completed) => {
+  const anothertask = new List(description, index, completed);
+  taskLs.push(anothertask);
+  updateLocalStorage();
+  showList();
+  console.log(`edited${index}`);
+};
+
+// edit item
+
+const toEdit = document.querySelector('.list-content');
+const editForm = document.querySelector('.edit-form');
+const form2 = document.querySelector('.form');
+const editItem = (tar) => {
+  if (tar.parentElement.classList.contains('edit')) {
+    form2.style.display = 'none';
+    editForm.style.display = 'block';
+    const editInput = document.querySelector('.editInput');
+    editInput.focus();
+    const itemId = tar.parentElement.parentElement.parentElement.classList.value;
+    const editItem = JSON.parse(localStorage.getItem('mylist'));
+
+    for (let i = 0; i < editItem.length; i += 1) {
+      if (!editItem[i]) {
+        console.log(' ');
+      } else if (editItem[i].index == itemId) {
+        editInput.value = editItem[i].description;
+        editForm.addEventListener('submit', (e) => {
+          e.preventDefault();
+          editItem[i].description = editInput.value;
+          editItem[i].completed = false;
+          editItem[i].index = itemId;
+          edits(
+            editItem[i].description,
+            editItem[i].index,
+            editItem[i].completed,
+          );
+          deleted(editItem[i].index); // delete item from ls
+          location.reload();
+        });
+      }
+    }
+  }
+};
```

Comment:
- [ ] When I edit a task and press the enter key, the task is deleted from the array instead of it being edited. I would advise you to use JavaScript's `dataset` to set an attribute on each task. After setting the attributes, you can loop through and get the particular task using the attribute. When you get the task, you can then change the value of the `description` key to the input value the user enters. After this, you can save the array to the `localStorage`. There are several ways to implement this feature, and the one I recommended is just one of the ways. Kindly ensure that the edit feature is implemented correctly. You can read this [article](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) to learn more about the dataset. 👍